data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/builder/KdfBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore.builder;
+
+import java.util.Objects;
+import tech.pegasys.artemis.bls.keystore.model.CryptoFunction;
+import tech.pegasys.artemis.bls.keystore.model.Kdf;
+import tech.pegasys.artemis.bls.keystore.model.KdfParam;
+import tech.pegasys.artemis.bls.keystore.model.SCryptParam;
+
+public final class KdfBuilder {
+  private CryptoFunction cryptoFunction;
+  private KdfParam param;
+
+  private KdfBuilder() {}
+
+  public static KdfBuilder aKdf() {
+    return new KdfBuilder();
+  }
+
+  public KdfBuilder withParam(final KdfParam param) {
+    this.param = param;
+    return this;
+  }
+
+  public Kdf build() {
+    Objects.requireNonNull(param);
+    cryptoFunction = param instanceof SCryptParam ? CryptoFunction.SCRYPT : CryptoFunction.PBKDF2;
```

Comment:
We shouldn't be doing `instanceof` here.  `KdfParam` should just have a `getCryptoFunction` method.

Reply:
implemented method to return correct function type.